asterius/rts/rts.gc.mjs

```diff
@@ -500,6 +501,23 @@ export class GC {
     for (const c of this.pinnedClosures) this.evacuateClosure(c);
     for (const[sp, c] of this.stablePtrManager.spt.entries())
       if (!(sp & 1)) this.stablePtrManager.spt.set(sp, this.evacuateClosure(c));
+
+    // Stage the movement of stable pointers. 
+    // Step 1: Move all the pointers
+    // Step 2: Update the pointer -> stablepointer mapping
+    // We cannot do this at the same time, since moving the pointer while
+    // we walk the ptr2stable map can yield an infinite loop:
+    // eg. (ptr:0 stablename: 42) --MOVE--> (ptr:1 stablename:42) --MOVE--> (ptr:2 stablename:42) ...
+    let ptr2stableMoved = new Map();
+    for (const[ptr, stable] of this.stableNameManager.ptr2stable.entries()) {
+      const ptrMoved = this.evacuateClosure(ptr);
+      ptr2stableMoved.set(ptrMoved, stable);
+    }
+    this.stableNameManager.ptr2stable.clear();
```

Comment:
We can use Object.seal instead of Object.freeze in stable name manager class, then directly assign the new mapping instead of clearing and adding updated entries.